napari/utils/colormaps/colormap.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+from scipy import interpolate
+from enum import auto
+
+from .colorbars import make_colorbar
+from .standardize_color import transform_color
+from ...utils.dataclass import dataclass, Property
+from ...utils.misc import StringEnum
+
+
+class ColormapInterpolationMode(StringEnum):
+    """INTERPOLATION: Interpolation mode for colormaps."""
+
+    LINEAR = auto()
+    ZERO = auto()
+
+
+@dataclass(events=True, properties=True)
+class Colormap:
```

Comment:
could be nice to add an `__iter__` method: 
```python
def __iter__(self):
    yield from (self.colors, self.controls, self.interpolation)
```
so that you can easily convert to a vispy colormap with:
```python
cm = napari.utils.Colormap(...)
vispy.colors.Colormap(*cm)
```

Reply:
btw, it takes 153 µs ± 1.25 µs on my computer to convert to a vispy colormap like that.  no biggie